File: TaskScheduler/TaskScheduler/Task.cs
namespace TaskScheduler

{   
    //This is the Task class. The components within this class is set to allow for sending error messages and the base
    //framework for how a Task is set up. This also sets the allowed lengths of the task's ID, name, and descriptions
    //to their maximum values.
    public class Task
    {
        //A constant integer that limits the length of the Task ID to 10 total integers' in length.
        public const int MaxTaskIdLength = 10;

        //This is a private string that is meant for showing the maximum length of the Task ID. It contains the
        //error message that the user receives.
        private static readonly string errorMsgInvalidTaskId =
            $"Task ID is invalid, please provide one no greater than "+
            $"{MaxTaskIdLength} characters.";

        //A constant integer that limits the length of the Task name to 10 total integers' in length.
        public const int MaxTaskNameLength = 20;

        //This is a private string that is meant for showing the maximum length of the task's name. It contains the
        //error message that the user receives.
        private static readonly string errorMsgInvalidTaskName =
            $"Task name invalid, please provide one no greater than " +
            $"{MaxTaskNameLength} characters.";

        //A constant integer that limits the length of the Task description to 10 total integers' in length.
        public const int MaxTaskDescriptionLength = 50;

        //This is a private string that is meant for showing the maximum length of the Task's description. It contains
        //the error message that the user receives.
        private static readonly string errorMsgInvalidTaskDescription =
            $"Task description invalid, please provide one no greater than" +
            $"{MaxTaskDescriptionLength} characters.";        

       //This is the constructor that assigns the string variables of id, name, and desc to their connecting
       //variables of Id, Name, and Description.
        public Task(string id, string name, string desc)
        {
            Id = id;
            Name = name;
            Description = desc;
        }
        
        //This is a string that starts off as empty and will contain the information for the ID.
        private string _id = string.Empty;
        
        //This is a getter and setter. This gets the ID, but sets it to make sure that it is not
        //exceeding the maximum number of allowed integers. If it does happen, then an invalid message is thrown
        //from the matching variable.
        public string Id
        {
            get => _id;
            private set
            {
                if ( string.IsNullOrEmpty(value) || value.Length > MaxTaskIdLength )
                {
                    throw new ArgumentException(errorMsgInvalidTaskId);
                }
                _id = value;
            }
        }

        //This is a string that starts off as empty and will contain the information for the name.
        private string _name = string.Empty; 

        //This is a getter and setter. This gets the name, but sets it to make sure that it is not
        //exceeding the maximum number of allowed integers. If it does happen, then an invalid message is thrown
        //from the matching variable.
        public string Name
        {
            get => _name;
            set
            {
                if(string.IsNullOrEmpty(value) || value.Length >MaxTaskNameLength)
                {
                    throw new ArgumentException(errorMsgInvalidTaskName);
                }
                _name = value;
            }
        }
        
        //This is a string that starts off as empty and will contain the information for the description.
        private string _description = string.Empty;
        
        //This is a getter and setter. This gets the description, but sets it to make sure that it is not
        //exceeding the maximum number of allowed integers. If it does happen, then an invalid message is thrown
        //from the matching variable.
        public string Description
        {
            get => _description;
            set
            {
                if (string.IsNullOrEmpty(value) || value.Length > MaxTaskDescriptionLength)
                {
                    throw new ArgumentException(errorMsgInvalidTaskDescription);
                }
                _description = value;
            }
        }
    }
}

File: Artifact2/Original/src/taskClassPackage/Task.java
package taskClassPackage;

public class Task {

	//This variable is specific to assigning an ID to a task and does not change.
	private String task_ID;
	
	//The name of the task and the task's description.
	private String name;
	private String desc;
	
	//A constructor that assigns the starting variables to themselves.
	public Task(String task_ID, String name, String desc)
	{
		if (task_ID == null || task_ID.length() > 10)
		{
			throw new IllegalArgumentException("Invalid Task ID");
		}
		
		if(name == null || name.length() > 20)
		{
			throw new IllegalArgumentException("Invalid Task Name");
		}
		
		if(desc == null || desc.length() > 50)
		{
			throw new IllegalArgumentException("Invalid Task Description");
		}
		this.task_ID = task_ID;
		this.name = name;
		this.desc = desc;
	}
	
	//A getter that gets the task ID's information.
	public String GetTaskID()
	{
		
		return task_ID;
	}
	
	//A getter that gets the name of the task.
	public String GetName()
	{
		
		return name;
	}
	
	//A setter that sets the task's name.
	public void SetName(String name)
	{
		if(name == null || name.length() >20)
		{
			throw new IllegalArgumentException("Invalid Task Name");
		}
		this.name = name;
	}
	
	//A getter that gets the description assigned to the task.
	public String GetDesc()
	{
		
		return desc;
	}
	
	//A setter that sets the description of the task.
	public void SetDesc(String desc)
	{
		if(desc == null || desc.length() > 50)
		{
			throw new IllegalArgumentException("Invalid Task Description");
		}
		
		this.desc = desc;
	}
}
